src/Driver.cpp

```diff
@@ -232,6 +235,57 @@ void Driver::build() {
   generator.clean();
 }
 
+void Driver::watch() {
+
+  build();
+
+  SourceWatcher config_watcher = SourceWatcher(config_file);
+  SourceWatcher watcher = SourceWatcher(files_patterns);
+
+  CppParser parser;
+  MarkdownGenerator generator(output);
+
+  for (;;){
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Those 500ms are chosen somewhat arbitrarily, for small projects I didn't see any noticeable load at this rate, for larger ones it could be too frequent however.

Reply:
Thoughts on changing this to 2000?

This is all anecdotal for a particular project where I use Doxide, but I see that at 500ms, `doxide watch` runs at about 9% of a single CPU, and at 2000ms a quarter of this (obviously). The real bottleneck anyway is `mkdocs` taking about 8 seconds to rebuild the documentation on a change, so it's not so critical that `doxide` responds so quickly in this particular case.

Reply:
I think 2000ms should be fine, as long as we choose a fixed delay it will always be a bit too much load for large projects and not as responsive as it could be for smaller ones so in doubt it is probably better to choose a larger delay.